### PR TITLE
logging and import cleanup in process_request.clj

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- logs backend response status and headers when debug is enabled
- cleans up unused imports
- makes logs consistently start with lower-case characters 

## Why are we making these changes?

Additional logging helps with debugging individual requests. Code cleanup is general hygiene.

### CODE REVIEW

File: waiter/src/waiter/process_request.clj [hunk 1]
```diff
@@ -507,6 +512,8 @@
   [post-process-async-request-response-fn _ instance-request-properties descriptor instance
    {:keys [uri] :as request} reason-map reservation-status-promise confirm-live-connection-with-abort
    request-state-chan {:keys [status] :as response}]
+  (when (utils/request->debug-enabled? request)
+    (log/info "backend response status:" (:status response)"and headers:" (:headers response)))
```
Comment (nsinkov) [inline waiter/src/waiter/process_request.clj]: space missing ```response)"and headers:"```

File: waiter/src/waiter/process_request.clj [hunk 2]
```diff
@@ -364,7 +369,7 @@
           (histograms/update! (metrics/service-histogram service-id "request-content-length") content-length)
           (statsd/inc! metric-group "request_content_length" content-length)))
       (catch Exception e
-        (log/error e "Unable to track content-length on request")))
```
Comment (nsinkov) [inline waiter/src/waiter/process_request.clj]: Just make all these capitalizations a separate PR. Search systematically. Then it will also cover the ```(log/error "there was a generic error``` from above

File: waiter/src/waiter/process_request.clj (hunk 1 above)
Comment (shamsimam) [reply waiter/src/waiter/process_request.clj]: Done.

File: waiter/src/waiter/process_request.clj (hunk 2 above)
Comment (shamsimam) [reply waiter/src/waiter/process_request.clj]: We are switching to lower-case. This PR changes the instances in this file.

Comment (nsinkov) [reply waiter/src/waiter/process_request.clj]: oops

Comment (nsinkov) [reply waiter/src/waiter/process_request.clj]: Let's open a task to replace ```log/(error|info|warn) .* ?"[A-Z]```

Comment (shamsimam) [reply waiter/src/waiter/process_request.clj]: Created: https://github.com/twosigma/waiter/issues/825
